File: HareQuestAPI/API/Models/ClassStudents.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Models
{
  public class ClassStudents
  {
    public Guid Id { get; set; }
    public Guid ParentId { get; set; }
    public Guid TeacherId { get; set; }
    public Guid ClassId { get; set; }
    [ForeignKey("ClassId")]
    public ClassGroups ClassGroup { get; set; }
    public string StudentName { get; set; }
    public string ClassCode { get; set; }
    public string ClassLevel { get; set; }
  }
}

File: HareQuest/HareQuest/SwiftDataModels/ClassStudents.swift
//
//  ClassStudents.swift
//  HareQuest
//
//  Created by Luke Hilliard on 04/05/2026.
//

import Foundation
import SwiftData

@Model
final class ClassStudents {
		@Attribute(.unique) var id: UUID
		var parentId: UUID
		var teacherId: UUID
		var classGroup: ClassGroup?
		var studentName: String
		var classCode: String
		var classLevel: ChildClass
	
		init(
				id: UUID = UUID(),
				parentId: UUID,
				teacherId: UUID,
				studentName: String,
				classCode: String,
				classLevel: ChildClass,
				classGroup: ClassGroup? = nil
		) {
				self.id = id
				self.parentId = parentId
				self.teacherId = teacherId
				self.studentName = studentName
				self.classCode = classCode
				self.classLevel = classLevel
				self.classGroup = classGroup
		}
}
